_posts/2019-09-19-uuid4-tiny.md

```diff
@@ -0,0 +1,49 @@
+---
+author: Daniel Mita
+layout: post
+title: UUID4::Tiny
+subtitle: Cryptographically secure version 4 UUIDs via the Linux getrandom() system call
+tags: perl uuid cryptography cpan module
+---
+
+[UUID::Tiny][uuid-tiny] is nice, but unfortunately due to its usage of
+rand() the version 4 UUIDs it provides are not cryptographically
+strong i.e. there exists the potential for the next value to be
+predicted. We opted to create a lightweight (core modules only!)
+alternative which uses a strong source of randomness.
+
+Linux kernel v3.17 introduced the system call getrandom() which
+returns a specified number of random bytes. It reads from the same
+source as /dev/urandom by default, without requiring the opening of
+a file handle (which removes an avenue for file descriptor
+exhaustion attacks).
```

Comment:
It's kind of the other way around.  It's that file descriptor exhaustion attacks could impede our ability to get reliable random numbers by preventing us from being able to open /dev/urandom in the first place.  By using a syscall we don't have that worry.